src/EditorFeatures/Core/Implementation/Classification/EditorClassifier.cs

```diff
@@ -83,6 +91,18 @@ internal static class EditorClassifier
             }
         }
 
+        private static void RemoveAdditiveSpans(List<ClassifiedSpan> spans)
+        {
+            for (var i = spans.Count - 1; i >= 0; i--)
+            {
+                var span = spans[i];
+                if (ClassificationTypeNames.AdditiveTypeNames.Contains(span.ClassificationType))
+                {
+                    spans.RemoveAt(i);
```

Comment:
ick.  N^2.  Any concern about that being a problem?  This could be done in a linear pass over the data.